src/config.ts

```diff
@@ -0,0 +1,131 @@
+                "databaseFilename": ""
+            },
+            "macFrameworkPath": [
+                "/System/Library/Frameworks",
+                "/Library/Frameworks"
+            ],
+            "compileCommands": "@DBPATH@"
+        },
+        {
+            "name": "Linux",
+            "includePath": [
+                "/usr/local/include",
+                "/usr/include",
+                "\${workspaceRoot}"
+            ],
+            "defines": [],
+            "intelliSenseMode": "clang-x64",
+            "browse": {
+                "path": [
+                    "/usr/local/include",
+                    "/usr/include",
+                    "\${workspaceRoot}"
+                ],
+                "limitSymbolsToIncludedHeaders": true,
+                "databaseFilename": ""
+            },
+            "compileCommands": "@DBPATH@"
+        },
+        {
+            "name": "Win32",
+            "includePath": [
+                "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include",
+                "\${workspaceRoot}"
+            ],
+            "defines": [
+                "_DEBUG",
+                "UNICODE"
+            ],
+            "intelliSenseMode": "msvc-x64",
+            "browse": {
+                "path": [
+                    "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include/*",
+                    "\${workspaceRoot}"
+                ],
+                "limitSymbolsToIncludedHeaders": true,
+                "databaseFilename": ""
+            },
+            "compileCommands": "@DBPATH@"
+        }
+    ],
+    "version": 3
+}`
+
+export class ConfigManager
+{
+    private readonly _workspaces: autoproj.Workspaces;
+    constructor(workspaces: autoproj.Workspaces)
+    {
+        this._workspaces = workspaces;
+    }
+    async setupPackage(pkgPath: string): Promise<boolean>
+    {
+        let ws = this._workspaces.folderToWorkspace.get(pkgPath); 
+        if (!ws)
+            return false;
+
+        let wsInfo: autoproj.WorkspaceInfo;
+        try {
+            wsInfo = await ws.info();
+        }
+        catch {
+            return false;
+        }
+
+        const pkgName = path.relative(ws.root, pkgPath)
+        let pkgModel = wsInfo.packages.get(pkgName);
+        if (!pkgModel || pkgModel.type != "Autobuild::CMake")
+            return false;
+
+        return this.writeCppProperties(pkgPath, pkgModel);
+    }
+    private writeCppProperties(pkgPath: string, pkgModel: autoproj.Package): boolean
+    {
+        const buildRelative = path.relative(pkgPath, pkgModel.builddir);
+        const dbPath = path.join("${workspaceRoot}", buildRelative, "compile_commands.json");
+        const fileData = C_CPP_PROPERTIES_JSON.replace(/@DBPATH@/g, dbPath);
+        const propertiesPath = path.join(pkgPath, ".vscode", "c_cpp_properties.json");
+        const options = {
+            mode: 0o644,
+            flag: 'w'
+        };
+        if (fs.existsSync(propertiesPath))
```

Comment:
if you have the courage, I would do an update logic that ensures that (1) there are entries for all three OSes and that (2) these entries have an up-to-date compileCommands field. It would ensure that the compile commands is up-to-date in all cases.

Reply:
Makes sense and should not be that difficult. But this will override user choice if for whatever reason he/she wants do provide his/her own compilation db. What about setting if the option is not there and leaving it as is? Not sure which is better..

Reply:
I really don't see a scenario where that would make any sense. If we do get this use-case, I would simply hide the auto-update into a configuration flag (which can be set per-folder, so we're covered).

The use-case I would like to cover is the case where someone does check in the property file into their VCS. I would expect to still have a valid setup in this case.

Reply:
Yeah, I agree.. I was over thinking/complicating this.